Running testcase: U2.TC-1
Verifikation af at Django pÃ¥ Docker fungerer pÃ¥ lokalmaskine
Linux og Mac
data-volume
mqtt-volume
data-volume
mqtt-volume
*** TEST STEPS ***
Running containers in interactive mode with logging
Please quit with CTRL-C when done viewing output
*** VERIFY NO ERRORS ***
Step 1/15 : FROM python:3.8.0-alpine
 ---> 204216b3821e
Step 2/15 : MAINTAINER Team Web, Janus Bo Andersen
 ---> Using cache
 ---> da61f29561e4
Step 3/15 : WORKDIR /usr/src/webinterface
 ---> Using cache
 ---> 50282d2928f0
Step 4/15 : ENV PYTHONDONTWRITEBYTECODE 1
 ---> Using cache
 ---> e4f005f1c751
Step 5/15 : ENV PYTHONUNBUFFERED 1
 ---> Using cache
 ---> 2b907d3d4c0f
Step 6/15 : RUN apk update     && apk add postgresql-dev gcc python3-dev musl-dev
 ---> Using cache
 ---> 884d7df53ed2
Step 7/15 : RUN pip install --upgrade pip
 ---> Using cache
 ---> cfb86b683141
Step 8/15 : COPY ./requirements.txt /usr/src/webinterface/requirements.txt
 ---> Using cache
 ---> 745f5226e473
Step 9/15 : RUN pip install -r requirements.txt
 ---> Using cache
 ---> 11dc8a9cfdb1
Step 10/15 : COPY ./entrypoint.sh /usr/src/webinterface/entrypoint.sh
 ---> Using cache
 ---> 267f178252b3
Step 11/15 : COPY . /usr/src/webinterface/
 ---> Using cache
 ---> 3a1a75644cc1
Step 12/15 : RUN chmod 777 /usr/src/webinterface/entrypoint.sh
 ---> Using cache
 ---> 547074c76777
Step 13/15 : RUN adduser -D nonrootuser
 ---> Using cache
 ---> 917197d57f22
Step 14/15 : USER nonrootuser
 ---> Using cache
 ---> 990d63ba636e
Step 15/15 : ENTRYPOINT ["/usr/src/webinterface/entrypoint.sh"]
 ---> Using cache
 ---> ba81996f9257

Successfully built ba81996f9257
Successfully tagged server-setup_webinterface:latest
Attaching to server-setup_db_1, server-setup_mqtt_1, server-setup_webinterface_1, server-setup_nginx_1
[36mdb_1            |[0m The files belonging to this database system will be owned by user "postgres".
[36mdb_1            |[0m This user must also own the server process.
[33mmqtt_1          |[0m 1574943327: mosquitto version 1.6.7 starting
[36mdb_1            |[0m 
[33mmqtt_1          |[0m 1574943327: Config loaded from /mosquitto/config/mosquitto.conf.
[32mwebinterface_1  |[0m Waiting for PostgreSQL service before starting Django...
[36mdb_1            |[0m The database cluster will be initialized with locale "en_US.utf8".
[33mmqtt_1          |[0m 1574943327: Opening ipv4 listen socket on port 1883.
[32mwebinterface_1  |[0m PostgreSQL started... Starting Django now.
[36mdb_1            |[0m The default database encoding has accordingly been set to "UTF8".
[33mmqtt_1          |[0m 1574943327: Opening ipv6 listen socket on port 1883.
[36mdb_1            |[0m The default text search configuration will be set to "english".
[36mdb_1            |[0m 
[36mdb_1            |[0m Data page checksums are disabled.
[36mdb_1            |[0m 
[36mdb_1            |[0m fixing permissions on existing directory /var/lib/postgresql/data ... ok
[36mdb_1            |[0m creating subdirectories ... ok
[36mdb_1            |[0m selecting dynamic shared memory implementation ... posix
[36mdb_1            |[0m selecting default max_connections ... 100
[36mdb_1            |[0m selecting default shared_buffers ... 128MB
[36mdb_1            |[0m selecting default time zone ... UTC
[36mdb_1            |[0m creating configuration files ... ok
[36mdb_1            |[0m running bootstrap script ... ok
[36mdb_1            |[0m performing post-bootstrap initialization ... sh: locale: not found
[36mdb_1            |[0m 2019-11-28 12:15:27.954 UTC [26] WARNING:  no usable system locales were found
[36mdb_1            |[0m ok
[36mdb_1            |[0m syncing data to disk ... initdb: warning: enabling "trust" authentication for local connections
[36mdb_1            |[0m You can change this by editing pg_hba.conf or using the option -A, or
[36mdb_1            |[0m --auth-local and --auth-host, the next time you run initdb.
[36mdb_1            |[0m ok
[36mdb_1            |[0m 
[36mdb_1            |[0m 
[36mdb_1            |[0m Success. You can now start the database server using:
[36mdb_1            |[0m 
[36mdb_1            |[0m     pg_ctl -D /var/lib/postgresql/data -l logfile start
[36mdb_1            |[0m 
[36mdb_1            |[0m waiting for server to start....2019-11-28 12:15:28.394 UTC [30] LOG:  starting PostgreSQL 12.0 on x86_64-pc-linux-musl, compiled by gcc (Alpine 8.3.0) 8.3.0, 64-bit
[36mdb_1            |[0m 2019-11-28 12:15:28.397 UTC [30] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
[36mdb_1            |[0m 2019-11-28 12:15:28.438 UTC [31] LOG:  database system was shut down at 2019-11-28 12:15:28 UTC
[36mdb_1            |[0m 2019-11-28 12:15:28.442 UTC [30] LOG:  database system is ready to accept connections
[36mdb_1            |[0m  done
[36mdb_1            |[0m server started
[36mdb_1            |[0m CREATE DATABASE
[36mdb_1            |[0m 
[36mdb_1            |[0m 
[36mdb_1            |[0m /usr/local/bin/docker-entrypoint.sh: ignoring /docker-entrypoint-initdb.d/*
[36mdb_1            |[0m 
[36mdb_1            |[0m waiting for server to shut down....2019-11-28 12:15:28.602 UTC [30] LOG:  received fast shutdown request
[36mdb_1            |[0m 2019-11-28 12:15:28.605 UTC [30] LOG:  aborting any active transactions
[36mdb_1            |[0m 2019-11-28 12:15:28.607 UTC [30] LOG:  background worker "logical replication launcher" (PID 37) exited with exit code 1
[36mdb_1            |[0m 2019-11-28 12:15:28.607 UTC [32] LOG:  shutting down
[36mdb_1            |[0m 2019-11-28 12:15:28.635 UTC [30] LOG:  database system is shut down
[36mdb_1            |[0m  done
[36mdb_1            |[0m server stopped
[36mdb_1            |[0m 
[36mdb_1            |[0m PostgreSQL init process complete; ready for start up.
[36mdb_1            |[0m 
[36mdb_1            |[0m 2019-11-28 12:15:28.724 UTC [1] LOG:  starting PostgreSQL 12.0 on x86_64-pc-linux-musl, compiled by gcc (Alpine 8.3.0) 8.3.0, 64-bit
[36mdb_1            |[0m 2019-11-28 12:15:28.724 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
[36mdb_1            |[0m 2019-11-28 12:15:28.724 UTC [1] LOG:  listening on IPv6 address "::", port 5432
[36mdb_1            |[0m 2019-11-28 12:15:28.732 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
[36mdb_1            |[0m 2019-11-28 12:15:28.768 UTC [41] LOG:  database system was shut down at 2019-11-28 12:15:28 UTC
[36mdb_1            |[0m 2019-11-28 12:15:28.773 UTC [1] LOG:  database system is ready to accept connections
[32mwebinterface_1  |[0m crontab: must be suid to work properly
[32mwebinterface_1  |[0m   adding cronjob: (b65a6d5cd88b4bd827cedf865d8a567d) -> ('0 0 * * *', 'database_poc.cron.Database_clean_up')
[32mwebinterface_1  |[0m crontab: must be suid to work properly
[32mwebinterface_1  |[0m Operations to perform:
[32mwebinterface_1  |[0m   Apply all migrations: admin, auth, contenttypes, sessions
[32mwebinterface_1  |[0m Running migrations:
[32mwebinterface_1  |[0m   Applying contenttypes.0001_initial... OK
[32mwebinterface_1  |[0m   Applying auth.0001_initial... OK
[32mwebinterface_1  |[0m   Applying admin.0001_initial... OK
[32mwebinterface_1  |[0m   Applying admin.0002_logentry_remove_auto_add... OK
[32mwebinterface_1  |[0m   Applying admin.0003_logentry_add_action_flag_choices... OK
[32mwebinterface_1  |[0m   Applying contenttypes.0002_remove_content_type_name... OK
[32mwebinterface_1  |[0m   Applying auth.0002_alter_permission_name_max_length... OK
[32mwebinterface_1  |[0m   Applying auth.0003_alter_user_email_max_length... OK
[32mwebinterface_1  |[0m   Applying auth.0004_alter_user_username_opts... OK
[32mwebinterface_1  |[0m   Applying auth.0005_alter_user_last_login_null... OK
[32mwebinterface_1  |[0m   Applying auth.0006_require_contenttypes_0002... OK
[32mwebinterface_1  |[0m   Applying auth.0007_alter_validators_add_error_messages... OK
[32mwebinterface_1  |[0m   Applying auth.0008_alter_user_username_max_length... OK
[32mwebinterface_1  |[0m   Applying auth.0009_alter_user_last_name_max_length... OK
[32mwebinterface_1  |[0m   Applying auth.0010_alter_group_name_max_length... OK
[32mwebinterface_1  |[0m   Applying auth.0011_update_proxy_permissions... OK
[32mwebinterface_1  |[0m   Applying sessions.0001_initial... OK
[32mwebinterface_1  |[0m Watching for file changes with StatReloader
[32mwebinterface_1  |[0m Performing system checks...
[32mwebinterface_1  |[0m 
[32mwebinterface_1  |[0m System check identified no issues (0 silenced).
[32mwebinterface_1  |[0m November 28, 2019 - 12:15:32
[32mwebinterface_1  |[0m Django version 2.2.7, using settings 'webinterface.settings'
[32mwebinterface_1  |[0m Starting development server at http://0.0.0.0:8000/
[32mwebinterface_1  |[0m Quit the server with CONTROL-C.
Gracefully stopping... (press Ctrl+C again to force)
